store/iavl/store.go

```diff
@@ -41,20 +42,36 @@ type Store struct {
 // LoadStore returns an IAVL Store as a CommitKVStore. Internally, it will load the
 // store's version (id) from the provided DB. An error is returned if the version
 // fails to load, or if called with a positive version on an empty tree.
-func LoadStore(db dbm.DB, id types.CommitID, lazyLoading bool) (types.CommitKVStore, error) {
-	return LoadStoreWithInitialVersion(db, id, lazyLoading, 0)
+func LoadStore(db dbm.DB, logger log.Logger, id types.CommitID, lazyLoading bool) (types.CommitKVStore, error) {
+	return LoadStoreWithInitialVersion(db, logger, id, lazyLoading, 0)
 }
 
 // LoadStoreWithInitialVersion returns an IAVL Store as a CommitKVStore setting its initialVersion
 // to the one given. Internally, it will load the store's version (id) from the
 // provided DB. An error is returned if the version fails to load, or if called with a positive
 // version on an empty tree.
-func LoadStoreWithInitialVersion(db dbm.DB, id types.CommitID, lazyLoading bool, initialVersion uint64) (types.CommitKVStore, error) {
+func LoadStoreWithInitialVersion(db dbm.DB, logger log.Logger, id types.CommitID, lazyLoading bool, initialVersion uint64) (types.CommitKVStore, error) {
 	tree, err := iavl.NewMutableTreeWithOpts(db, defaultIAVLCacheSize, &iavl.Options{InitialVersion: initialVersion})
 	if err != nil {
 		return nil, err
 	}
 
+	if tree.IsUpgradeable() && logger != nil {
+		logger.Info(
+			"module's storage upgrade is in progress while loading",
```

Comment:
```suggestion
			"Upgrading IAVL storage for faster queries + execution on live state. This may take awhile",
```

Reply:
Oh wait, does this get called for each module independently?

Reply:
Yes, each module has a separate store accessible by key [[doc](https://docs.cosmos.network/master/core/store.html)].

It looks like we create a separate prefix db for each module [here](https://github.com/osmosis-labs/cosmos-sdk/blob/f4329814320de3f0f6170f9347053c28afc60831/store/rootmulti/store.go#L866).

Then, when running a node, we can see multiple logs stating the upgrade. We could also propagate the store key to be able to log it. What do you think?

Reply:
Waiting on your reply about the store key before I make this change

Reply:
What do the logs look like right now?

Reply:
Haven't changed it yet